compiler/src/iree/compiler/Codegen/LLVMCPU/Passes.cpp

```diff
@@ -667,15 +662,28 @@ static void addLowerToLLVMPasses(OpPassManager &modulePassManager,
   if (enableAArch64SME) {
     modulePassManager.addPass(mlir::arm_sme::createVectorLegalizationPass());
     FunctionLikeNest(modulePassManager)
+        .addPredicatedPass(
+            clForceArmStreaming,
+            [] {
+              // Scalable dispatches forced to use streaming mode may not need
+              // ZA (so that is enabled separately for dispatch regions that
+              // need it below).
+              return mlir::arm_sme::createEnableArmStreamingPass(
+                  mlir::arm_sme::ArmStreamingMode::StreamingLocally,
+                  mlir::arm_sme::ArmZaMode::Disabled,
+                  /*ifRequiredByOps=*/false,
+                  /*ifContainsScalableVectors=*/true);
+            })
         .addPass(createCanonicalizerPass)
         .addPass(createCSEPass)
         .addPass(mlir::createArithToArmSMEConversionPass)
         .addPass(mlir::createConvertVectorToArmSMEPass)
-        .addPass([]() {
+        .addPass([] {
```

Comment:
Why is this added twice? Once by default and once behind a flag?

Reply:
They are configured differently and enable different things. One enables streaming mode for dispatches that contain scalable vectors (if the `iree-llvmcpu-force-arm-streaming` is set). One enables streaming mode and ZA for dispatches that contain ArmSME ops.